### PR TITLE
[PM-2855] Add Settings to Enable Autofill Overlay

### PR DESCRIPTION
**Pull request by cagonzalezcs**
## Type of change

```
- [ ] Bug fix
- [x] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

The changes for this PR address two specific concerns:

- The autofill settings page within the extension now presents logic that allows end users the ability to enable/disable autofill as desired. It also has two appearance settings that allow the user the ability to show the overlay list on click of the icon, or on focus of a valid input field. 
- We also introduce behavior that allows specifically Chromium browser users to OPT-IN to disable the browser's default autofill behavior.

NOTE: None of the actual overlay behavior is in place here. Just the settings that facilitate turning the overlay on/off.

## Code changes

- **apps/browser/src/_locales/en/messages.json:** Content to be translated representing the settings labels for the autofill overlay.
- **apps/browser/src/autofill/utils/autofill-overlay.enum.ts:** Adding an enum for the two ways that the autofill overlay can present itself. Used within the settings component as well as the overlay content script
- **apps/browser/src/manifest.json:** Adding an optional permission for `privacy` to facilitate an opt-in override of Chrome's autofill behavior
- **apps/browser/src/manifest.v3.json:** Adding an optional permission for `privacy` to facilitate an opt-in override of Chrome's autofill behavior
- **apps/browser/src/platform/browser/browser-api.ts:** Adding methods that facilitate overriding default autofill behavior for Chrome browsers (opt-in)
- **apps/browser/src/popup/settings/autofill.component.html:** Updates to the autofill settings component to ensure that users can set the autofill overlay on/off and update the appearance of the overlay itself. These will only appear if the launch darkly feature flag is set.
- **apps/browser/src/popup/settings/autofill.component.ts:** Adding the server config observer to this component to conditionally show the autofill overlay settings. Also adding code required to view and update the autofill overlay settings.
- **libs/common/src/abstractions/settings.service.ts** Adding a public method `getEnableAutoFillOverlay` to the settings service.
- **libs/common/src/platform/abstractions/state.service.ts:** Adding methods required to get and set the values that enable the autofill overlay and control the overlay's appearance
- **libs/common/src/platform/models/domain/account.ts:** Adding methods that allow for validating if the overlay is set within a user account and the manner in which the overlay should appear.
- **libs/common/src/platform/services/state.service.ts:** Adding methods required to get and set the values that enable the autofill overlay and control the overlay's appearance. The logic for this is very similar to the methodology used to set the autofill on load setting and it's default values.
- **libs/common/src/services/settings.service.ts:** Adding a method for identifying if the autofill overlay is currently enabled

## Screenshots

https://github.com/bitwarden/clients/assets/16629865/03b52b70-0952-4839-8655-ca861aecee01

### CODE REVIEW

File: apps/browser/src/autofill/utils/autofill-overlay.enum.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+const AutofillOverlayVisibility = {
+  Off: 0,
+  OnButtonClick: 1,
+  OnFieldFocus: 2,
+} as const;
```
Comment (Hinton) [inline apps/browser/src/autofill/utils/autofill-overlay.enum.ts:5]: Why const over enum?

File: apps/browser/src/popup/settings/autofill.component.ts [hunk 2]
```diff
@@ -1,17 +1,27 @@
...
 @Component({
   selector: "app-autofill",
   templateUrl: "autofill.component.html",
 })
 export class AutofillComponent implements OnInit {
+  canOverrideBrowserAutofillSetting = false;
+  defaultBrowserAutofillDisabled = false;
+  isAutoFillOverlayFlagEnabled = false;
+  autoFillOverlayVisibility: number;
+  autoFillOverlayVisibilityOptions: any[];
```
Comment (Hinton) [inline apps/browser/src/popup/settings/autofill.component.ts:24]: **note**: These could be made protected.

**Review by Hinton — APPROVED** [review]
I think this looks good from platform and tech-lead perspective. The settings is a bit clunky but not sure how we can improve that right now.

File: apps/browser/src/popup/settings/autofill.component.ts (hunk 2 above)
Comment (cagonzalezcs) [reply apps/browser/src/popup/settings/autofill.component.ts:24]: Thanks for the suggestion, I've updated these to be protected within https://github.com/bitwarden/clients/pull/6514

File: apps/browser/src/autofill/utils/autofill-overlay.enum.ts (hunk 1 above)
Comment (cagonzalezcs) [reply apps/browser/src/autofill/utils/autofill-overlay.enum.ts:5]: This was discussed on Slack, and is being further discussed with the team.

**Pull request merged**